command/agentproxyshared/cache/lease_cache.go

```diff
@@ -625,13 +625,56 @@ func (c *LeaseCache) storeStaticSecretIndex(ctx context.Context, req *SendReques
 		return err
 	}
 
-	// TODO: We need to also update the cache for the token's permission capabilities.
-	// TODO: for this we'll need: req.Token, req.URL.Path
-	// TODO: we need to build a NEW index, with a hash of the token as the ID
+	capabilitiesIndex, err := c.retrieveOrCreateTokenCapabilitiesEntry(req.Token)
+	if err != nil {
+		c.logger.Error("failed to cache the proxied response", "error", err)
+		return err
+	}
+
+	path := getStaticSecretPathFromRequest(req)
+
+	// Extra caution -- avoid potential nil
```

Comment:
Back when I was toying around with this initially (before the code was distinguishing properly between static and dynamic secrets), I managed to hit this code path when this was interacting with dynamic secrets and got a panic exactly here. Of course, that "shouldn't" happen, especially since the code can determine if a static secret is static, but given that experience, I felt a little safer adding adding the 'belt and braces'. That being said, open to removing it.